File: app/src/main/java/jp/hazuki/yuzubrowser/utils/view/CopyableTextView.java
package jp.hazuki.yuzubrowser.utils.view;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import jp.hazuki.yuzubrowser.utils.ClipboardUtils;

public class CopyableTextView extends AppCompatTextView implements TextView.OnLongClickListener {
    public CopyableTextView(Context context) {
        this(context, null);
    }

    public CopyableTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        setOnLongClickListener(this);
    }

    @Override
    public boolean onLongClick(View v) {
        ClipboardUtils.setClipboardText(getContext().getApplicationContext(), getText().toString());
        return true;
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/utils/view/CopyableTextView.kt
package jp.hazuki.yuzubrowser.utils.view

import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.util.AttributeSet
import android.view.View

import jp.hazuki.yuzubrowser.utils.extensions.setClipboardWithToast

class CopyableTextView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : AppCompatTextView(context, attrs), View.OnLongClickListener {

    init {
        setOnLongClickListener(this)
    }

    override fun onLongClick(v: View): Boolean {
        context.setClipboardWithToast(text.toString())
        return true
    }
}
